### DISCUSSION TITLE
How do class work?

### DISCUSSION THREAD

**Discussion by thatdumbdevit** [FabricMC/fabric-api#1429] [Mod Dev Support] [answered]
Hi,
I'm still approaching the developing world, so there's something I don't understand
I often see this:
```
import net.minecraft.class_1267;
import net.minecraft.class_1297;
import net.minecraft.class_1299;
import net.minecraft.class_1309;
import net.minecraft.class_1559;
import net.minecraft.class_1657;
import net.minecraft.class_1972;
import net.minecraft.class_2246;
import net.minecraft.class_2338;
import net.minecraft.class_3218;
import net.minecraft.class_3532;
```
or even this:

```
public boolean maxNearbyMonsters(class_3218 world, class_1657 player) {
        int max;
        if (world.method_8407() == class_1267.field_5805) {
            max = 1;
        } else if (world.method_8407() == class_1267.field_5802) {
            max = 2;
        } else if (world.method_8407() == class_1267.field_5807) {
            max = 3;
        } else {
            max = 1;
        }

```
How does "class_number" or "method_number" work?

**Comment by haykam821** [comment, marked as answer]
These are intermediary names for unmapped classes and methods. They function exactly like normal classes and methods, but do not have a useful name or package to refer to them (such as `DynamicRegistryManager` in place of `class_5455`).

**Reply by thatdumbdevit** [reply]
Is there a list or something for the references of the various numbers?

**Reply by haykam821** [reply]
There is a [class list](https://maven.fabricmc.net/docs/yarn-21w16a+build.12/allclasses-index.html) in the Yarn Javadoc.

**Reply by thatdumbdevit** [reply]
The class list you sent me is for Minecraft 21w16a (1.17 Snapshot). Can you send me the 1.16.5 version?

**Reply by Chocohead** [reply]
The list for 1.16.5 is [here](https://maven.fabricmc.net/docs/yarn-1.16.5+build.8/allclasses-index.html).

I would ask where you're getting the sources in Intermediary from in the first place? If you're decompiling the Minecraft jar you'd be better running the `genSources` task which will use the Yarn names rather than Intermediary.

**Reply by thatdumbdevit** [reply]
I was decompiling a mod that I found on Internet. Also, thanks for the advice. I always use `genSources` on VSCode, but in that moment I was not using any IDE, I was using jadx-gui-1.2.0, a program that just allow to see the decompiled content of a Jar file, but nothing else. Another question, is [this](https://github.com/InventivetalentDev/minecraft-assets) the official decompiled Jar of Minecraft? If it isn't, can you link me the official one?

**Reply by naritanara** [reply]
I would recommend you to check out this [wiki page on obfuscation and mappings](https://fabricmc.net/wiki/tutorial:mappings), as it looks like you don't know either concept. Read it from start to end, and I think it will all become clear.

**Reply by thatdumbdevit** [reply]
As I said, I'm still approaching the developers world, so thank you, that will be really useful for me!
